src/sprightly.ts

```diff
@@ -58,28 +83,42 @@ function sprightly(
     ...defaultOptions,
     ...options,
   }
-  const entryPointAbsolutePath = path.resolve(path.normalize(entryPoint.trim()))
-  if (options.cache && cache.has(entryPointAbsolutePath))
-    return cache.get(entryPointAbsolutePath) as string
+
+  console.log(path.resolve(entryPoint))
+  const entryPointAbsolutePath = path.resolve(path.normalize(entryPoint))
+  if (options.cache && cache.has(entryPointAbsolutePath)) {
+    return cache.get(entryPointAbsolutePath)!
+  }
+
   try {
     const file = readFileSync(entryPointAbsolutePath).toString()
     const parsedFile = parse(file, data, options)
-    if (options.cache && !cache.has(entryPointAbsolutePath))
+
+    if (options.cache && !cache.has(entryPointAbsolutePath)) {
       cache.set(entryPointAbsolutePath, parsedFile)
+    }
+
     return parsedFile
   } catch (error) {
-    if (error instanceof KeyNotFoundError)
-      throw `Key "${error.key}" not found in "data" at ${entryPointAbsolutePath}`
-    if (error instanceof PathNotFoundError)
-      throw `Component ${error.component} was not found in ${entryPointAbsolutePath}`
+    if (error instanceof KeyNotFoundError) {
+      throw new KeyNotFoundError(error.key, entryPoint)
+    }
+    if (error instanceof PathNotFoundError) {
+      throw new PathNotFoundError(error.component, entryPoint)
+    }
     throw error
```

Comment:
Here, instead of doing all the above, check if the error is thrown by `sprightlyAsync`, and if so, rethrow it using the yet-to-implement custom error class about a path being not found. If it is not, then just re-throw the error directly.